### DISCUSSION TITLE
Holding down navigation keys causes scroll animation to act strangely

### DISCUSSION THREAD

**Discussion by dotfrag** [folke/snacks.nvim#270] [Q&A]
Maybe it's just me, because I see a lot of people like to use scroll animation in neovim, but I could never get it to work.

This is not a snacks only problem, and is the main reason I never used any animation plugin.

When holding down `C-e`, `C-y`, `C-d`, `C-u`, `C-f`, `C-b`, animations become "choppy". It's either looking stuck, then suddenly jumping, or moving in a "slow-fps" mode. Sorry I don't have better words to describe it.

Please tell me I'm not the only one encountering this, with every single animation plugin in neovim?

Are there any known workarounds?

I want to be clear that my intention is not to bash on any plugin authors, just ranting a bit because I would really like for scroll animations in neovim to work smoothly.

**Comment by folke** [comment]
I just added spamming detection, so this should help for your use-case 😅

When spamming is detected, animations are temporarily disabled until the user behaves again.

I never encountered that issue, since I never spam keys like that.

Do let me know if it's better this way.

**Reply by dotfrag** [reply]
I often hold down the keys when I know I wanna navigate from let's say 20% of the file to 80%, or when opening long files (think log files), and I would guess you never use `C-e` `C-y`? As the only point of those is surely to hold them down? :p

Soo c-e and c-y are definitely better (no animation judging from the commit msg), but c-d and c-u is bit meh. I can't tell you if it's better or worse than before.

Before it was choppy but at least I could see where the cursor was.

Now it's not choppy, but it's _very_ jumpy. i.e. animation starts, then suddenly jumps many lines below/above, throwing my perception of where I am in the file off.

I don't know how much this helps or if it;s possible, but the ideal logic for me is the following:

1. User starts holding C-d
2. Start animation and scroll down one C-d distance (as if C-d was only pressed and released)
3. When animation is done, check if C-d is still held down
4. If still held down, repeat  from 2.

This effectively reduces the total amount of line scrolled during a certain held down duration, but is both smooth _and_ you can still see where you are.

PS. appreciate it you are trying to help!

**Reply by folke** [reply]
Like I said, animations just stop so it behaves as it would without animations.
Won't spend any more time on this. Just disable scroll if you don't like it, or stop spamming keys.

**Reply by dotfrag** [reply]
No problem, thanks!
